### PR TITLE
Add selection from casa cases to hearing types

### PR DESCRIPTION
**Pull request by haydenrou**
### What github issue is this PR for, if any?
This PR addresses half of #927 
Due to #929 not being complete, I cannot complete the second half of this PR (adding the association and selection of judge names), so I will submit that separately later.

### What changed, and why?
An association and column has been added between casa cases and hearing types, and a selection for this in the create page has been added. I have shown it on the casa case show and index page too.

### Screenshots please :)
addhearingtypes

### CODE REVIEW

File: spec/models/casa_case_spec.rb [hunk 1]
```diff
@@ -75,7 +76,7 @@
 
     context "when volunteer has no case assignments" do
       it "returns all cases in volunteer's organization" do
-        expect(described_class.available_for_volunteer(volunteer)).to eq [casa_case2, casa_case3, casa_case1]
+        expect(described_class.available_for_volunteer(volunteer).ids).to eq [casa_case2.id, casa_case3.id, casa_case1.id]
```
Comment (haydenrou) [inline spec/models/casa_case_spec.rb:79]: specerror
Some reason I'm getting this error if I don't change this - so I amended to fix.

File: app/views/casa_cases/_form.html.erb [hunk 2]
```diff
@@ -12,6 +12,16 @@
         <% end %>
       </div>
 
+      <div class="field form-group">
+        <%= form.label :hearing_type_id %>
+        <%= form.collection_select(
+          :hearing_type_id,
+          HearingType.active.for_organization(current_user.casa_org),
```
Comment (haydenrou) [inline app/views/casa_cases/_form.html.erb:19]: stackerror
Getting a `stack level too deep`error here if using `current_organization` for some reason. Tried to debug briefly with Jeanine in Slack with no luck as of yet. Open to suggestions.

Comment (compwron) [reply app/views/casa_cases/_form.html.erb:19]: hmm weird. probably a factories problem. We should fix this
I made https://github.com/rubyforgood/casa/issues/1166 to track it, feel free to pick it up or not  :)

File: spec/models/casa_case_spec.rb (hunk 1 above)
Comment (compwron) [reply spec/models/casa_case_spec.rb:79]: If this screenshot included the section under "Diff" maybe I could tell what was up (or maybe not) but this seems fine for now :)

**Review by compwron — APPROVED** [review]

File: spec/models/casa_case_spec.rb (hunk 1 above)
Comment (haydenrou) [reply spec/models/casa_case_spec.rb:79]: Argh, it must've been to do with some of the other tests I hadn't fixed yet. I can't replicate it. Sorry! Would you like me to revert that part in a separate PR or shall we leave it?

Comment (compwron) [reply spec/models/casa_case_spec.rb:79]: Revert in separate PR would be great :)

**Pull request merged**